gpt_index/indices/base.py

```diff
@@ -194,6 +194,16 @@ def refresh(
 
         return refreshed_documents
 
+    @property
+    def additional_query_context(self) -> Dict[str, Any]:
```

Comment:
out of curiosity, how is this different than preprocess_query / should we deprecate preprocess_query then?

Reply:
I wanted to separate out:
1. args that we should allow overriding at query time, "configurations" type arguments, like similarity_top_k
2. args that should not be, "state" or "service" type arguments, like vector store.

For 2, ideally we have a proper abstraction that captures all of the required context (e.g. the Index class). But to avoid making that big refactor, I'm going with explicitly adding a new `additional_query_context` property to the Index class. so that I can pass it to the query runner.